src/util/io_limiter.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::option::Option;
+use std::io::{Result, Write};
+
+use rocksdb::RateLimiter;
+
+const REQUEST_PRIORITY: u8 = 1;
+const REFILL_PERIOD: i64 = 100 * 1000;
+const FARENESS: i32 = 10;
+pub const DEFAULT_SNAP_MIN_BYTES_PER_TIME: u64 = 128 * 1024;
+pub const DEFAULT_SNAP_MAX_BYTES_PER_TIME: u64 = 2 * 1024 * 1024;
+pub const DEFAULT_SNAP_MAX_BYTES_PER_SEC: u64 = 20 * 1024 * 1024;
+
+pub struct IOLimiter {
+    inner: Option<RateLimiter>,
```

Comment:
Seems not reasonable to create an io limiter without a rate limiter, why not control it outside?

Reply:
Sure, i will correct this.